runtime/sema/checker.go

```diff
@@ -998,13 +998,18 @@ func (checker *Checker) convertRestrictedType(t *ast.RestrictedType) Type {
 		// TODO: also include interface conformances's members
 		//   once interfaces can have conformances
 
-		for name := range restrictionInterfaceType.Members {
+		restrictionInterfaceType.Members.Foreach(func(name string, member *Member) {
 			if previousDeclaringInterfaceType, ok := memberSet[name]; ok {
 
 				// If there is an overlap in members, ensure the members have the same type
 
-				memberType := restrictionInterfaceType.Members[name].TypeAnnotation.Type
-				previousMemberType := previousDeclaringInterfaceType.Members[name].TypeAnnotation.Type
+				memberType := member.TypeAnnotation.Type
+
+				prevMemberType, ok := previousDeclaringInterfaceType.Members.Get(name)
+				if !ok {
+					return
```

Comment:
AFAICS the member lookup should never fail, otherwise that's an implementation error, i.e. `previousDeclaringInterfaceType.Members` wasn't populated properly. Maybe `panic` with an unreachable error

Reply:
Yeah agree. Added `panic` for all such similar places.